hazelcast/src/test/java/com/hazelcast/spi/impl/SlowOperationDetectorTest.java

```diff
@@ -0,0 +1,589 @@
+            return null;
+        }
+    }
+
+    private static class SlowEntryProcessorChild extends SlowEntryProcessor {
+        private SlowEntryProcessorChild(int sleepSeconds) {
+            super(sleepSeconds);
+        }
+
+        @Override
+        public Object process(Map.Entry<String, String> entry) {
+            try {
+                TimeUnit.SECONDS.sleep(sleepSeconds);
+            } catch (InterruptedException ignored) {
+            }
+            return null;
+        }
+    }
+
+    private static class NestedSlowEntryProcessor implements EntryProcessor<String, String> {
+
+        final IMap<String, String> map;
+        final int sleepSeconds;
+
+        private NestedSlowEntryProcessor(IMap<String, String> map, int sleepSeconds) {
+            this.map = map;
+            this.sleepSeconds = sleepSeconds;
+        }
+
+        @Override
+        public Object process(Map.Entry<String, String> entry) {
+            map.executeOnKey(DEFAULT_KEY, new SlowEntryProcessor(sleepSeconds));
+            return null;
+        }
+
+        @Override
+        public EntryBackupProcessor<String, String> getBackupProcessor() {
+            return null;
+        }
+    }
+
+    private static class SlowRunnable implements PartitionSpecificRunnable {
+
+        private final int sleepSeconds;
+        private final int partitionId;
+
+        private SlowRunnable(int sleepSeconds, int partitionId) {
+            this.sleepSeconds = sleepSeconds;
+            this.partitionId = partitionId;
+        }
+
+        @Override
+        public void run() {
+            sleepSeconds(sleepSeconds);
+        }
+
+        @Override
+        public int getPartitionId() {
+            return partitionId;
+        }
+    }
+
+    private static class SlowOperation extends AbstractOperation {
+
+        private final int sleepSeconds;
+
+        private SlowOperation(int sleepSeconds) {
+            this.sleepSeconds = sleepSeconds;
+        }
+
+        private SlowOperation(int sleepSeconds, int partitionId) {
+            this.sleepSeconds = sleepSeconds;
+            this.setPartitionId(partitionId);
+        }
+
+        @Override
+        public void run() throws Exception {
+            sleepSeconds(sleepSeconds);
+        }
+    }
+
+    private static class NestedSlowOperationOnSamePartition extends AbstractOperation {
+
+        private final IMap<String, String> map;
+        private final int sleepSeconds;
+
+        private NestedSlowOperationOnSamePartition(IMap<String, String> map, int partitionId, int sleepSeconds) {
+            this.map = map;
+            this.sleepSeconds = sleepSeconds;
+
+            this.setPartitionId(partitionId);
+        }
+
+        @Override
+        public void run() throws Exception {
+            map.executeOnKey(DEFAULT_KEY, new SlowEntryProcessor(sleepSeconds));
+        }
+    }
+
+    private static class NestedSlowOperationOnDifferentThreadPools extends AbstractOperation {
```

Comment:
I don't understand the meaning of this class, because you are not doing anything with threadpools. So what is this operation doing?